src/ckb-daemon/structures.h

```diff
@@ -247,7 +248,7 @@ typedef struct {
     // Color dithering in use
     char dither;
     // Flag to check, if large macros should be sent delayed
-    char delay;
+    uint delay;
```

Comment:
Maybe there is another possible reason for the crashes: the Firmware in the KB itself.
The reason why I implemented the original delay was the overflow of the buffer in the unix User-device-driver in the direction from ckb-daemon to user-input. When I had longer macros, some of the keys were dropped - independent of the keystroke-frequency.
When I tested this, there was only 1 keystroke type ahead. With a long macro (I tested with hundreds and some thousand chars) and pressing the G-Key 3 times, the macro string appeared 2 times.
If I pressed G-key 2 times and a 3rd time when the 2nd macro was executing, a third macro string appeared and so on.
IMHO this is a behaviour of the USB user-driver in linux, but I didn't check it in the code. But if I'm right, there might be no response to the KB via USB. Maybe there is the reason for the failure. So if we can assure that the USB protocol isn't affected by our delays, we come a step further: We get your choice No 4.
On the other hand: Who needs repeatedly sending of very, very slow macros?
I'm not curious, but it's killing me, if I do not know something...

My suggestion: Let's implement the  rest of the UI, bring it to the testing stage and wait for comments. Maybe this is a problem only on debian / ubuntu based systems with older kernels and with  mint18 the effect is gone - who knows...